internal/pkg/skuba/kubernetes/jobs.go

```diff
@@ -60,7 +59,7 @@ func CreateAndWaitForJob(client clientset.Interface, name string, spec batchv1.J
 			} else {
 				if job.Status.Succeeded > 0 {
 					klog.V(1).Infof("job %s executed successfully", name)
-					return nil
+					return DeleteJob(client, name)
```

Comment:
we've talked about this offline @ereslibre , seems that we only have to delete if the job succeeded, but better if you can double check

Reply:
This one is tricky. If the job succeeded and the cleanup of the succeeded job failed, should we return an error? The main task, that was removing the node succeeded though.

Not a big deal, but interesting nevertheless.

Reply:
let's not change the behaviour in this PR, just informing that the deletion failed.
I will add a TODO so this can be revisited later